src/probability/integration.lean

```diff
@@ -118,4 +120,132 @@ lintegral_mul_eq_lintegral_mul_lintegral_of_independent_measurable_space
   (measurable_iff_comap_le.1 h_meas_f) (measurable_iff_comap_le.1 h_meas_g) h_indep_fun
   (measurable.of_comap_le le_rfl) (measurable.of_comap_le le_rfl)
 
+/-- This proves that if `f` and `g` are independent and almost everywhere measurable,
+   then `E[f * g] = E[f] * E[g]` (slightly generalizing
+   `lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun`). -/
+lemma lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun' [measurable_space α] {μ : measure α}
+  {f g : α → ℝ≥0∞} (h_meas_f : ae_measurable f μ) (h_meas_g : ae_measurable g μ)
+  (h_indep_fun : indep_fun f g μ) :
+  ∫⁻ a, (f * g) a ∂μ = ∫⁻ a, f a ∂μ * ∫⁻ a, g a ∂μ :=
+begin
+  rcases h_meas_f with ⟨f',f'_meas,f'_ae⟩,
+  rcases h_meas_g with ⟨g',g'_meas,g'_ae⟩,
+  have fg_ae : f * g =ᵐ[μ] f' * g' := f'_ae.mul g'_ae,
+  rw [lintegral_congr_ae f'_ae, lintegral_congr_ae g'_ae, lintegral_congr_ae fg_ae],
+  apply lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun f'_meas g'_meas,
+  exact h_indep_fun.ae_eq f'_ae g'_ae
```

Comment:
We have API for `ae_measurable`, which should be used instead of those `rcases`. The measurable function is called `h_meas_f.mk f`, its measurability is `h_meas_f.measurable_mk` and the a.e. equality is `h_meas_f.ae_eq_mk`.
```suggestion
  have fg_ae : f * g =ᵐ[μ] (h_meas_f.mk f) * (h_meas_g.mk g),
    from h_meas_f.ae_eq_mk.mul h_meas_g.ae_eq_mk,
  rw [lintegral_congr_ae h_meas_f.ae_eq_mk, lintegral_congr_ae h_meas_g.ae_eq_mk,
    lintegral_congr_ae fg_ae],
  apply lintegral_mul_eq_lintegral_mul_lintegral_of_indep_fun h_meas_f.measurable_mk
    h_meas_g.measurable_mk,
  exact h_indep_fun.ae_eq h_meas_f.ae_eq_mk h_meas_g.ae_eq_mk
```

Reply:
Looking into it. Any reason why `f` is not implicit in `ae_measurable.mk`? the term `h_meas_f.mk f` feels a bit redundant.

Reply:
The docstring of `ae_measurable.mk` says : " f is explicit in the definition to make sure that it shows in pretty-printing."